error-budgets/SLOs-SLIs.md

```diff
@@ -0,0 +1,64 @@
+# Definitions of SLOs and SLIs for packit
+
+If you are not familiar with SLOs and SLIs, please check [README.md](./README.md) first.
+
+## How do others do it?
+
+The CKI team has a pretty comprehensive set of
+[SLOs](https://cki-project.org/docs/hacking/rfcs/cki-004-slo/#created-slos-based-on-the-feedback).
+Instead of taking the same route, let's start small and slowly add more and
+more objectives and indicators over time.
+
+Testing Farm team has [Service level
+expectations](https://docs.google.com/document/d/1iHK3v_tCK4w-F82ZvUEO3xHcNv78tkR31w3oF-zpi1Q/edit#heading=h.26gd46u8e15q)
+and a pretty slick
+[dashboard](http://grafana-latest-osci-infra.cloud.privileged.psi.redhat.com/d/VXwH27XMk/rhel-infrastructure-and-service-health?orgId=1).
+
+Cockpit also uses this framework and the team has [a nice dashboard as well](http://grafana-latest-osci-infra.cloud.privileged.psi.redhat.com/d/VgtdtiwGk/rhel-infrastructure-and-service-health?orgId=1).
+
+## Packit SLOs
+
+SLOs are based on [discussions with our stakeholders](users-expectations.md).
+
+Let's define job statuses first to get a clear understanding of these objectives:
+
+- successful run: a job finished and everything is awesome
+- failure: the job finished and wasn't successful: the PR needs to be fixed
+- error: infrastructure problems prevented the job to complete
+
+### 99% of builds finish with success/failure by 24 hours
+
+The most core functionality is to run COPR builds for PRs. We want to be sure
+those builds either pass or fail and no error interrupts the build process.
+
+The problem is that some builds take minutes and some hours so it's hard to
+design this objective to make it more proactive - that we'd know sooner that
+something is wrong.
+
+### 95% of test runs finish with success/failure by 24 hours
+
+Similar as builds but since Testing Farm is outside of our control, let's lower
+the percentage.
+
+### Changes to GitHub PRs receive a status update within 15 seconds in 99% of cases
+
+It can be frustrating (for us and our users) when we push changes to our PRs
+while no statuses are being set. Let's make a deadline for packit to set these.
+
+## Packit SLIs
+
+If we want to track our SLOs, we need to start measuring different aspects of our workflow.
+
+- Number of builds that finished with an error
+
+- Number of unfinished builds that are in progress for more than 24 hours
+
+- Number of test runs that finished with an error
+
+- Number of PRs handled by packit with no commit status from packit in >15s
+
+- Time it takes packit to set the initial status
```

Comment:
These indicators seem to me to be derivatives.

Shouldn't we have some more basic ones which can be combined to derive and measure the objectives? Like: number of builds started, number of builds finished with successful/failure/error, build duration, etc...